### PR TITLE
Make the plugin compileable again

### PR DESCRIPTION
**Pull request by NotMyFault**
The change proposed makes the plugin compilable again because the default branch can't be compiled at the moment.

Just for my understanding, is there a particular reason why you kept merging pull requests, even though these broke your default branch and failed the CI checks @dheid ?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -21,8 +21,8 @@
         <revision>3.2.0</revision>
         <changelist>999999-SNAPSHOT</changelist>
         <gitHubRepo>jenkinsci/postbuildscript-plugin</gitHubRepo>
-        <java.level>8</java.level>
```
Comment (NotMyFault) [inline pom.xml:24]: Is obsolete and is no longer configured on a plugin-pom level.

File: pom.xml [hunk 2]
```diff
@@ -21,8 +21,8 @@
...
-        <jenkins.version>2.359</jenkins.version>
+        <jenkins.version>2.414.1</jenkins.version>
```
Comment (NotMyFault) [inline pom.xml:24]: Aligns with the baseline recommendation

**Comment by dheid** [comment]
Thanks for fixing this. Some build jobs failed although everything went through in the past. So I decided to collect the updates and fix it all at once before a new release.

File: pom.xml [hunk 3]
```diff
@@ -21,8 +21,8 @@
...
+        <hpi.compatibleSinceVersion>3.0.0</hpi.compatibleSinceVersion>
```
Comment (NotMyFault) [inline pom.xml:25]: Inlined the property from the obsolete maven-hpi-plugin block

File: pom.xml [hunk 4]
```diff
@@ -101,7 +101,7 @@
     </contributors>
 
     <scm>
-        <connection>scm:git:git://github.com/${gitHubRepo}.git</connection>
+        <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
```
Comment (NotMyFault) [inline pom.xml:104]: The former syntax is deprecated and emits a warning

File: pom.xml [hunk 5]
```diff
@@ -112,7 +112,6 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>matrix-project</artifactId>
-            <version>808.v5a_b_5f56d6966</version>
```
Comment (NotMyFault) [inline pom.xml:115]: Shipped by the BOM declared below

**Comment by NotMyFault** [comment]
CI checks are succesful 🚀

**Pull request merged**